x-pack/plugins/security_solution/public/detections/components/take_action_dropdown/index.tsx

```diff
@@ -0,0 +1,247 @@
+/* Todo: Uncomment case action after getAddToCaseAction is split into action and modal
+import {
+    ACTION_ADD_TO_CASE
+} from '../alerts_table/translations';
+import { useGetUserCasesPermissions, useKibana } from '../../../common/lib/kibana';
+import { useInsertTimeline } from '../../../cases/components/use_insert_timeline';
+import { addToCaseActionItem } from './helpers'; */
+import { useEventFilterAction } from '../alerts_table/timeline_actions/use_event_filter_action';
+import { useHostIsolationAction } from '../host_isolation/use_host_isolation_action';
+import { CHANGE_ALERT_STATUS } from './translations';
+import { getFieldValue } from '../host_isolation/helpers';
+import type { Ecs } from '../../../../common/ecs';
+import { Status } from '../../../../common/detection_engine/schemas/common/schemas';
+
+interface ActionsData {
+  alertStatus: Status;
+  eventId: string;
+  eventKind: string;
+  ruleId: string;
+  ruleName: string;
+}
+
+export const TakeActionDropdown = React.memo(
+  ({
+    detailsData,
+    ecsData,
+    handleOnEventClosed,
+    isHostIsolationPanelOpen,
+    loadingEventDetails,
+    nonEcsData,
+    onAddEventFilterClick,
+    onAddExceptionTypeClick,
+    onAddIsolationStatusClick,
+    refetch,
+    timelineId,
+  }: {
+    detailsData: TimelineEventsDetailsItem[] | null;
+    ecsData?: Ecs;
+    handleOnEventClosed: () => void;
+    isHostIsolationPanelOpen: boolean;
+    loadingEventDetails: boolean;
+    nonEcsData?: TimelineNonEcsData[];
+    refetch: (() => void) | undefined;
+    onAddEventFilterClick: () => void;
+    onAddExceptionTypeClick: (type: ExceptionListType) => void;
+    onAddIsolationStatusClick: (action: 'isolateHost' | 'unisolateHost') => void;
+    timelineId: string;
+  }) => {
+    /* Todo: Uncomment case action after getAddToCaseAction is split into action and modal
+    const casePermissions = useGetUserCasesPermissions();
+    const { timelines: timelinesUi } = useKibana().services;
+    const insertTimelineHook = useInsertTimeline;
+    */
+    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+    const actionsData = useMemo(
+      () =>
+        [
+          { category: 'signal', field: 'signal.rule.id', name: 'ruleId' },
+          { category: 'signal', field: 'signal.rule.name', name: 'ruleName' },
+          { category: 'signal', field: 'signal.status', name: 'alertStatus' },
+          { category: 'event', field: 'event.kind', name: 'eventKind' },
+          { category: '_id', field: '_id', name: 'eventId' },
+        ].reduce<ActionsData>(
+          (acc, curr) => ({
+            ...acc,
+            [curr.name]: getFieldValue({ category: curr.category, field: curr.field }, detailsData),
+          }),
+          {} as ActionsData
+        ),
+      [detailsData]
+    );
+
+    const alertIds = useMemo(() => [actionsData.eventId], [actionsData.eventId]);
+    const isEvent = actionsData.eventKind === 'event';
+
+    const isEndpointAlert = useMemo((): boolean => {
```

Comment:
there's actually a helper function written in `public/common/utils/endpoint_alert_check.ts` that you could use